js/plates-model/plate.js

```diff
@@ -179,6 +181,75 @@ export default class Plate extends PlateBase {
     return count
   }
 
+  markIslands () {
+    // DFS-based algorithm which calculates area of continents and mark small ones as islands.
+    const queue = []
+    const visited = {}
+    const area = {}
+    const continentId = {}
+
+    const calcAreaOfContinent = function () {
+      while (queue.length > 0) {
+        const field = queue.pop()
+        const cId = continentId[field.id]
+        area[cId] += field.area
+        field.forEachNeighbour(neighbour => {
+          if (neighbour.isContinent && !visited[neighbour.id]) {
+            queue.push(neighbour)
+            visited[neighbour.id] = true
+            continentId[neighbour.id] = cId
+          }
+        })
+      }
+    }
+
+    this.forEachField(field => {
+      field.island = false
+      if (field.isContinent && !visited[field.id]) {
+        queue.push(field)
+        visited[field.id] = true
+        continentId[field.id] = field.id
+        area[continentId[field.id]] = 0
+        calcAreaOfContinent()
+      }
+    })
+
+    this.forEachField(field => {
+      const cId = continentId[field.id]
+      if (cId !== undefined && area[cId] < MAX_ISLAND_SIZE) {
+        field.island = true
+      }
+    })
+  }
+
+  mergeIsland (island, collidingField) {
+    const perfectPosition = island.absolutePos
+    let bestFieldId = null
+    let minDist = Infinity
+
+    for (let adjId of collidingField.adjacentFields) {
+      const adjField = this.adjacentFields.get(adjId)
+      if (adjField) {
+        if (adjField.absolutePos.distanceTo(perfectPosition) < minDist && adjField.neighboursCount() > 1) {
```

Comment:
What's the purpose of the `adjField.neighboursCount() > 1` check? To make sure you don't collect islands in some kind of spike?

Reply:
Exactly. Narrow spikes don't look very realistic and it's more likely that we end up with a plate with some holes. I've added a comment to clarify that too.